### PR TITLE
changres to favourites List

### PR DESCRIPTION
**Pull request by Chukwukaodinaka**

### CODE REVIEW

File: sep6/cypress.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (rafall404) [inline sep6/cypress.json:1]: Is this necessary?

File: sep6/cypress/integration/1-getting-started/todo.spec.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+/// <reference types="cypress" />
+
```
Comment (rafall404) [inline sep6/cypress/integration/1-getting-started/todo.spec.js:2]: does this even test something on our project?

File: sep6/public/index.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
 <!--    <meta name="google-signin-scope" content="profile email">-->
 <!--    <meta name="google-signin-client_id" content="656774347674-tf2md9b56kuosa26savdln8poc97bdnm.apps.googleusercontent.com">-->
-    <script src="https://apis.google.com/js/api:client.js"></script>
+      <script type="text/javascript" src="https://apis.google.com/js/api:client.js?onLoad=onGoogleScriptLoaded"></script>
```
Comment (rafall404) [inline sep6/public/index.html]: refactor code

File: sep6/src/auth/index.js [hunk 4]
```diff
@@ -50,7 +50,7 @@
                         errorCallback(error)
                     })
                 } else {
-                    window.gapi.auth2.getAuthInstance().grantOfflineAccess({'redirect_uri': 'postmessage'}).then(function (response) {
+                     window.gapi.auth2.getAuthInstance().grantOfflineAccess({'redirect_uri': 'postmessage'}).then(function (response) {
```
Comment (rafall404) [inline sep6/src/auth/index.js]: refactor

File: sep6/src/store/index.js [hunk 5]
```diff
@@ -6,8 +6,8 @@ Vue.use(Vuex)
 
 const APIKEY = '4f8f2699713f7c3cc1758f5f2f2ed5e7'
 
-// const backendUrl = "http://localhost:8888"
-const backendUrl = "https://sep6-back-end-an6w7okvaa-lz.a.run.app"
+const backendUrl = "http://localhost:8888"
+// const backendUrl = "https://sep6-back-end-an6w7okvaa-lz.a.run.app"
```
Comment (rafall404) [inline sep6/src/store/index.js]: you need to put back the real URL if you want to merge on production

File: sep6/src/store/index.js [hunk 6]
```diff
@@ -69,7 +71,49 @@ const actions = {
...
+        axios.get(`${backendUrl}/favorites/${userId}`)
+            .then(response => {
+                commit('SET_FAVOURITE_LIST', response.data)
+            })
+    },
+    addFavourite({commit},{userId, movieId}) {
+        let url =`${backendUrl}/favorites/${userId}/${movieId}`
+        axios.post(url)
+            .then(response => {
+                console.log(response.data + "@@@@@@@@@@")
```
Comment (rafall404) [inline sep6/src/store/index.js]: no logs on production

File: sep6/src/store/index.js [hunk 7]
```diff
@@ -20,7 +20,9 @@ const state = {
     searchResultList: [],
     movieDetails: {},
     trendingList: [],
-    movie: {}
+    movie: {},
+    favouriteList:[],
+    isFavourite:false
```
Comment (rafall404) [inline sep6/src/store/index.js]: why wouldn't we keep favourite in movie object?

File: sep6/src/store/index.js [hunk 8]
```diff
@@ -69,7 +71,49 @@ const actions = {
...
+    //Favourites------------------------------------------------
+
+    registerUser({commit},{userId,username}) {
+        let url = `${backendUrl}/users/register/${userId}/${username}`
+        axios.post(url)
+            .then(response => {
+                commit('',response.data)
+            })
+            .catch(error =>{
+                console.log(error)
```
Comment (rafall404) [inline sep6/src/store/index.js:83]: proper handling needed

File: sep6/src/view/SearchPage.vue [hunk 9]
```diff
@@ -28,6 +28,7 @@ export default {
   },
   computed: {
     movieList() {
+      console.log(this.$store.state.searchResultList)
```
Comment (rafall404) [inline sep6/src/view/SearchPage.vue:31]: no need to merge it

File: sep6/src/view/MoviePage.vue [hunk 10]
```diff
@@ -158,6 +169,23 @@ export default {
...
+      return currentUser
+    },
+    addFav(){
+      this.$store.state.isFavourite = true
+      let VueInstance = this
+      this.$store.dispatch("addFavourite",{userId: this.getUserLoggedIn(),movieId: VueInstance.movieId})
+    },
+    removeFav(){
+      this.$store.state.isFavourite = false
+      console.log(this.isFav)
```
Comment (rafall404) [inline sep6/src/view/MoviePage.vue]: no need to merge it

File: sep6/src/store/index.js [hunk 11]
```diff
@@ -90,6 +134,23 @@ const mutations = {
...
+        if(status == "200"){
+            state.favouriteList.push(movieId)
+        }
+    },
+    DELETE_FAVOURITE_LIST(state,status,movieId){
+        if(status == "200") {
+            state.favouriteList = state.favouriteList.filter(id => id != movieId)
+        }
+    },
+    IS_FAVOURITE(state,response){
```
Comment (rafall404) [inline sep6/src/store/index.js]: Again, no need for that. Isfavourite can be returned in movie details object

File: sep6/src/view/FavouritesPage.vue [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+import MovieList from "./MovieList";
+import {getAuth, onAuthStateChanged} from "firebase/auth";
+export default {
+  name: "FavouritesPage",
+  components:{
+    MovieList
+  },
+  computed:{
+    movieList(){
+      console.log(JSON.stringify(this.$store.state.favouriteList))
```
Comment (rafall404) [inline sep6/src/view/FavouritesPage.vue]: I am no sure if I can find all the log you have made but you need to at least TRY to not push them

File: sep6/src/view/MovieList.vue [hunk 13]
```diff
@@ -11,14 +11,14 @@
     </div>
 
     <div class="movie_view row align-md-baseline align-center" ref="movie_list">
-      <v-col class="movie_item col-lg-2 col-md-4 col-12" v-model="movieList"
+      <v-col class="movie_item col-lg-3 col-md-4 col-12" v-model="movieList"
```
Comment (rafall404) [inline sep6/src/view/MovieList.vue:14]: why 3 now?

File: sep6/src/view/MovieList.vue [hunk 14]
```diff
@@ -11,14 +11,14 @@
...
              v-for="(movie, _key)   in movieList" v-on:click="openModal(movie)"  @mouseover="activeOver(_key)" @mouseleave="removeOver(_key)" :key="_key">
         <img id="movie_item_pic"  v-bind:src="`${movie.poster}`">
         <div >
           {{ movie.title }}
         </div>
         <div>
-          {{ "movie.release_date" }}
+          {{ movie.release_date }}
```
Comment (rafall404) [inline sep6/src/view/MovieList.vue:21]: there is no such attribute

File: sep6/src/view/MoviePage.vue [hunk 15]
```diff
@@ -141,15 +146,21 @@ export default {
 
   }),
   computed: {
+    isFav(){
```
Comment (rafall404) [inline sep6/src/view/MoviePage.vue:149]: no need for that if isFaurourite would be in movie object

File: sep6/src/view/MoviePage.vue [hunk 16]
```diff
@@ -141,15 +146,21 @@ export default {
...
     directors() {
       return this.$store.state.trendingList.slice(10, 12)
     }
   },
-  mounted() {
-    this.$store.dispatch("getMovieDetails", {movieId: this.movieId});
+  async mounted() {
+    let VueInstance = this
+    await this.$store.dispatch("getMovieDetails", {movieId: VueInstance.movieId});
+    await this.$store.dispatch("inFavourite", {userId: VueInstance.getUserLoggedIn(),movieId: VueInstance.movieId})
```
Comment (rafall404) [inline sep6/src/view/MoviePage.vue]: can go into movie details

File: sep6/cypress/integration/2-advanced-examples/actions.spec.js [hunk 17]
```diff
@@ -0,0 +1,299 @@
+/// <reference types="cypress" />
+
```
Comment (rafall404) [inline sep6/cypress/integration/2-advanced-examples/actions.spec.js]: do we need it?

File: sep6/cypress/integration/2-advanced-examples/aliasing.spec.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
+/// <reference types="cypress" />
+
```
Comment (rafall404) [inline sep6/cypress/integration/2-advanced-examples/aliasing.spec.js]: what does it test in our app?

File: sep6/cypress/support/index.js [hunk 19]
```diff
@@ -0,0 +1,20 @@
...
+//
+// You can read more here:
+// https://on.cypress.io/configuration
+// ***********************************************************
+
+// Import commands.js using ES2015 syntax:
+import './commands'
+
+// Alternatively you can use CommonJS syntax:
+// require('./commands')
```
Comment (rafall404) [inline sep6/cypress/support/index.js:20]: no need for MP4 videos in our project

**Review by rafall404 — CHANGES_REQUESTED** [review]

File: sep6/cypress.json (hunk 1 above)
Comment (Chukwukaodinaka) [reply sep6/cypress.json:1]: Yes. cypress test can't run without cypress.json

File: sep6/cypress/integration/1-getting-started/todo.spec.js (hunk 2 above)
Comment (Chukwukaodinaka) [reply sep6/cypress/integration/1-getting-started/todo.spec.js:2]: no.. but its proof of concept for Continous Integration until we start testing

**Pull request closed without merging**